compose/service/values/expr.go

```diff
@@ -0,0 +1,113 @@
+package values
+
+import (
+	"context"
+	"fmt"
+	"github.com/PaesslerAG/gval"
+	"github.com/cortezaproject/corteza-server/compose/types"
+)
+
+func makeInvalidExprErr(field *types.ModuleField, expr string, err error) types.RecordValueError {
+	return types.RecordValueError{
+		Kind:    "valueExpression",
+		Message: fmt.Sprintf("invalid expression %q: %v", expr, err.Error()),
+		Meta:    map[string]interface{}{"field": field.Name},
+	}
+}
+func makeExprEvalErr(field *types.ModuleField, expr string, err error) types.RecordValueError {
+	return types.RecordValueError{
+		Kind:    "valueExpression",
+		Message: fmt.Sprintf("failed to evaluate formula expression %q: %v", expr, err.Error()),
+		Meta:    map[string]interface{}{"field": field.Name},
+	}
+}
+
+func makeValueExprIncompErr(field *types.ModuleField) types.RecordValueError {
+	return types.RecordValueError{
+		Kind:    "evaluatedValueIncompatible",
+		Message: "evaluated results incompatible",
+		Meta:    map[string]interface{}{"field": field.Name},
+	}
+}
+
+func parser() gval.Language {
+	return gval.Full()
+}
+
+// Expression evaluates expression in ModuleField.Expressions.Value and
+// assigns results to the record on that field
+func Expression(ctx context.Context, m *types.Module, r *types.Record, old *types.Record, rve *types.RecordValueErrorSet) {
+	var (
+		exprParser = parser()
+
+		scope = make(map[string]interface{})
+
+		reserved = map[string]bool{
+			"new": true,
+			"old": true,
+		}
```

Comment:
You could ditch this and just overwrite the reserved indexes with those values at the end